src/codeu/chat/client/BroadCastReceiver.java

```diff
@@ -0,0 +1,127 @@
+package codeu.chat.client;
+
+import codeu.chat.common.ConversationSummary;
+import codeu.chat.common.Message;
+import codeu.chat.common.NetworkCode;
+import codeu.chat.util.Serializers;
+import codeu.chat.util.connections.Connection;
+import codeu.chat.util.connections.ConnectionSource;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Created by rsharif on 3/23/17.
+ */
+public class BroadCastReceiver extends Thread{
+
+    private ConnectionSource mySource;
+    private OutputStream out;
+    private BroadcastEvent myResponse;
+    private boolean alive;
+    private InputStream in;
+    private int lastType;
+    private AtomicBoolean receivedResponse;
+
+    // A broadcast event will be fired whenever a new broadcast is pushed to the client.
+    @FunctionalInterface
+    public interface BroadcastEvent { void onBroadcast(Message message); }
+
+    public BroadCastReceiver(ConnectionSource mySource) {
+        this.mySource = mySource;
+        this.alive = true;
+        this.receivedResponse = new AtomicBoolean(false);
+    }
+
+
+    @Override
+    public void run() {
+
+        try (
+                final Connection myConnection = this.mySource.connect()
+        ) {
+            in = myConnection.in();
+            out = myConnection.out();
+
+            while (alive) {
+
+
+                if (!receivedResponse.get()) {
+
+                    int type = Serializers.INTEGER.read(in);
+
+                    if (type == NetworkCode.NEW_BROADCAST) {
+                        Message message = Message.SERIALIZER.read(in);
+                        if (myResponse != null) myResponse.onBroadcast(message);
+                        // todo send a broadcast response to inform server that broadcast was received
+                    } else if (type == NetworkCode.JOIN_CONVERSATION_RESPONSE) {
+                        receivedResponse.set(true);
+                    } else {
+                        this.lastType = type;
+                        receivedResponse.set(true);
+                        Thread.yield();
+                    }
+
+                }
+            }
+
+        } catch (IOException exc) {
+            if(alive) System.out.println("Error connecting with broadcaster");
+        }
+
+    }
+
+    public void joinConversation(ConversationSummary old, ConversationSummary newCon){
```

Comment:
To join a conversation, I only really care about the Uuid for the conversation they wish to join so I figured there was no need for the client and server to send an entire conversation instance